libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -1133,55 +1131,78 @@ string ABIFunctions::abiDecodingFunctionValueType(Type const& _type, bool _fromM
 string ABIFunctions::abiDecodingFunctionArray(ArrayType const& _type, bool _fromMemory)
 {
 	solAssert(_type.dataStoredIn(DataLocation::Memory), "");
-	solAssert(!_type.isByteArray(), "");
 
 	string functionName =
 		"abi_decode_" +
 		_type.identifier() +
 		(_fromMemory ? "_fromMemory" : "");
 
-	solAssert(!_type.dataStoredIn(DataLocation::Storage), "");
```

Comment:
Shouldn't we keep this?

Reply:
There is an assert `solAssert(_type.dataStoredIn(DataLocation::Memory), "");` just before this one, so I think it is redundant

Reply:
ah right!